File: service/src/main/java/by/rekuts/travelagency/service/impl/ReviewServiceImpl.java
package by.rekuts.travelagency.service.impl;

import by.rekuts.travelagency.domain.Review;
import by.rekuts.travelagency.repository.ReviewRepository;
import by.rekuts.travelagency.service.ReviewService;
import lombok.RequiredArgsConstructor;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@RequiredArgsConstructor
@Service
public class ReviewServiceImpl implements ReviewService {

    private final ReviewRepository reviewRepository;

    @Transactional
    @Override
    public void insert(Review review) {
        reviewRepository.save(review);
    }

    @Transactional
    @Override
    public void update(Review review) {
        reviewRepository.save(review);
    }

    @Transactional
    @Override
    public void delete(int id) {
        reviewRepository.deleteById(id);
    }

    @Transactional
    @Override
    public List<Review> getList(Specification<Review> specification) {
        return reviewRepository.findAll(specification);
    }
}

File: service/src/main/scala/by/rekuts/travelagency/service/impl/ReviewServiceImpl.scala
package by.rekuts.travelagency.service.impl

import java.util

import by.rekuts.travelagency.domain.Review
import by.rekuts.travelagency.repository.{ReviewRepository, Specification}
import by.rekuts.travelagency.service.ReviewService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class ReviewServiceImpl extends ReviewService {
  @Autowired
  private var reviewRepository: ReviewRepository = _

  @Transactional
  override def insert(review: Review): Unit = reviewRepository.insert(review)

  @Transactional
  override def update(review: Review): Unit = reviewRepository.update(review)

  @Transactional
  override def delete(id: Int): Unit = reviewRepository.delete(id)

  @Transactional
  override def getList(specification: Specification[_]): util.List[Review] = reviewRepository.getList(specification)
}
